lib/manage/do_rename.c

```diff
@@ -94,7 +94,7 @@ int M_do_rename(int n, const char *old, const char *new)
 	}
 
 	if (G_strcasecmp(list[n].element[0], "cell") == 0) {
-	    char colr2[50];
+	    char colr2[GNAME_MAX];
 
 	    sprintf(colr2, "colr2/%s", G_mapset());
```

Comment:
Quick glance at this: `G_mapset()` may return a string of `GMAPSET_MAX` (= GNAME_MAX). This may potentially case segfault, no?
Perhaps a solution like:
https://github.com/OSGeo/grass/blob/38c084ad0b45bc2a46e333eaf4ec5ecabde5d315/lib/vector/Vlib/map.c#L494
could be considered?

Reply:
Right, but then other instances in `lib/manage` need to be updated as well to use `G_snprintf()`. I will change it accordingly. Maybe also use
```
	    char colr2[GNAME_MAX + 6];
```
?

Reply:
`colr2` is supposed to be of max `GNAME_MAX` size, isn't it. If so, then `GNAME_MAX + 6` wouldn't be the best choice.

Re.: G_snprintf(), I believe the C standard `snprintf()` is nowadays well supported and preferable. Remember our discussion on https://github.com/OSGeo/grass/pull/1406#discussion_r594719601? I have even a local branch for replace usage and remove/deprecate G_snprintf().

Reply:
Or isn’t `colr2 ` size of `GNAME_MAX + GNAME_MAX `?

Reply:
It sure was a quick glance. Sorry for the noice, but now I looked a bit closer at what's what:

`char colr2[GNAME_MAX + 6];` should be enough. No real need for snprintf() in this case.